### DISCUSSION TITLE
Password reset for user

### DISCUSSION THREAD

**Discussion by MrTrebron** [sqlpage/SQLPage#77] [Q&A] [answered]
Hi there,
any suggestions how to handle / implement a "forgot password" feature?

Thank
Norbert

**Comment by lovasoa** [comment, marked as answer]
Hello ! Usually password reset is implemented by sending an email with a secret URL containing a password reset token to the user. SQLPage itself cannot send emails. What you can do in SQLPage with standard SQL is generating a random *password reset* token, store it in the database, and use an external system to send the email.

```sql
INSERT INTO password_reset_tokens(token, username) VALUES (sqlpage.random_string(32), :username)
```

You can read: https://cheatsheetseries.owasp.org/cheatsheets/Forgot_Password_Cheat_Sheet.html

If you really are using a postgres database and you really need to, you can send an email directly from postgres using an extension such as [pgMail](https://github.com/captbrando/pgMail).
